x-pack/filebeat/input/awss3/config.go

```diff
@@ -100,9 +100,9 @@ func (c *config) Validate() error {
 	}
 
 	if c.AWSConfig.FIPSEnabled && c.NonAWSBucketName != "" {
-		return errors.New("fips_enabled cannot be used with a non-AWS S3 bucket.")
+		return errors.New("fips_enabled cannot be used with a non-AWS S3 bucket")
 	}
-	if c.PathStyle && c.NonAWSBucketName == "" {
+	if c.PathStyle && c.NonAWSBucketName == "" && c.QueueURL == "" {
 		return errors.New("path_style can only be used when polling non-AWS S3 services")
```

Comment:
maybe edit this error message as well to include queue_url?

Reply:
I think this has to be a little different now.

`path_style can only be used when using non-AWS S3 services or SQS/SNS QueueURL`